basics_of_statistics_in_sst-core/statistics.md

```diff
@@ -0,0 +1,670 @@
+stats_example, INT64_r_clock_ticks, , Histogram, 8000, 0, 0, 9, 5, 2, 44, 284, 2, 8, 8, 0, 0, 3, 5
+stats_example, INT64_f_clock_ticks, , Histogram, 9000, 0, 0, 9, 5, 2, 36, 204, 2, 9, 9, 0, 0, 5, 4
+stats_example, INT64_r_clock_ticks, , Histogram, 9000, 0, 0, 9, 5, 2, 45, 285, 2, 9, 9, 0, 0, 4, 5
+stats_example, INT64_f_clock_ticks, , Histogram, 10000, 0, 0, 9, 5, 2, 45, 285, 2, 10, 10, 0, 0, 5, 5
+stats_example, INT64_r_clock_ticks, , Histogram, 10000, 0, 0, 9, 5, 2, 45, 285, 2, 10, 10, 0, 0, 5, 5
+stats_example, INT64_f_clock_ticks, , Histogram, 11000, 0, 0, 9, 5, 2, 45, 285, 2, 10, 10, 0, 0, 5, 5
+stats_example, INT64_r_clock_ticks, , Histogram, 11000, 0, 0, 9, 5, 2, 45, 285, 2, 10, 10, 0, 0, 5, 5
+stats_example, INT64_f_clock_ticks, , Histogram, 11000, 0, 0, 9, 5, 2, 45, 285, 2, 10, 10, 0, 0, 5, 5
+stats_example, INT64_r_clock_ticks, , Histogram, 11000, 0, 0, 9, 5, 2, 45, 285, 2, 10, 10, 0, 0, 5, 5
+```
+
+### Accumulator and Histogram Printed to a CSV File
+
+The final example is intended to demonstrate that components can be associated with different types of statistics.  In this case, the first statistic is defined as an accumulator, the second as a histogram.  The results are printed to a CSV file.
+```
+# Execute from the command line with the command:
+#   sst StatsConfig.py
+#
+import sst
+
+# Initialize local variables.
+#
+debug = "0"         # debug level
+                    # 0 = FATAL, 1 = WARN, 2 = INFO, 3 = DEBUG, 4 = TRACE, 5 = ALL
+
+componentName0 = "stats_example"
+
+# Set global params for statistics.
+#
+sst.setStatisticLoadLevel(7)
+sst.setStatisticOutput("sst.statOutputCSV",
+                       {"filepath":"./StatsOutput.txt",
+                        "separator":", "})
+
+# Component definition
+#
+obj = sst.Component(componentName0, "statsexample.StatsComponent")
+obj.addParams({
+    "clock"      : "1GHz",
+    "debug"      : debug
+    })
+
+# Enable the statistics.
+#
+sst.enableStatisticForComponentName(componentName0,
+                                    "INT64_f_clock_ticks", 
+                                   {"type":"sst.AccumulatorStatistic",
+                                    "rate":"1ns"})
+sst.enableStatisticForComponentName(componentName0,
+                                    "INT64_r_clock_ticks", 
+                                   {"type":"sst.HistogramStatistic",
+                                    "rate":"1ns",
+                                    "minvalue":"0",
+                                    "binwidth":"5",
+                                    "numbins":"2"})
+```
+
+The expected results are listed below.  The values for each statistic are interspersed so further analysis of an individual statistic will require the file values to be parsed in some manner.  In addition, the statistic name has not been included in the results.  Therefore, some knowledge of the statistic type will be required when parsing data for individual statistics. 
```

Comment:
The stat name not being included is a bug in SST (thanks for finding it!). It appears to be linked to using the enableStatisticByComponentName() function with CSV output, we are tracking it down.